tracer/src/Datadog.Trace/Configuration/TracerSettingsSnapshot.cs

```diff
@@ -48,11 +48,47 @@ internal partial class TracerSettingsSnapshot : SettingsSnapshotBase
             var newValue = settings.IntegrationsInternal.Settings[i];
             var oldValue = Integrations.Settings[i];
             var integrationName = newValue.IntegrationNameInternal;
-            RecordIfChanged(telemetry, string.Format(ConfigurationKeys.Integrations.Enabled, integrationName), oldValue.EnabledInternal, newValue.EnabledInternal);
+
+            RecordChanges();
+
+            void RecordChanges()
```

Comment:
Can't you use the `GetKeys()` approach here? IMO we could just use that and remove the string.format/string/create approach entirely (as we're guaranteed to only be working with existing integration ids here)